src/capture_RPiHQ.cpp

```diff
@@ -251,294 +266,253 @@ void writeToLog(int val)
-		bin = 1;  // for me the best default value
+		bin = 3;
 	}
-	if (bin < 1) 	{
-		bin = 1; // for me the best default value
+	else if (bin < 1) 	{
+		bin = 1;
 	}
 
-	// Check for binning 1x1 is selected
 	// https://www.raspberrypi.com/documentation/accessories/camera.html#raspistill
 	// Mode   Size         Aspect Ratio  Frame rates  FOV      Binning/Scaling
-  // 0      automatic selection
+	// 0      automatic selection
 	// 1      2028x1080    169:90        0.1-50fps    Partial  2x2 binned
 	// 2      2028x1520    4:3           0.1-50fps    Full     2x2 binned      <<< bin==2
 	// 3      4056x3040    4:3           0.005-10fps  Full     None            <<< bin==1
 	// 4      1332x990     74:55         50.1-120fps  Partial  2x2 binned      <<< else 
 	//
-  // todo: please change gui description !
+  // TODO: please change gui description !
 
 	if (bin==1)	{
-		// Select binning 1x1 (4060 x 3056 pixels)
-		roi = "--mode 3 ";
+		command += " --mode 3";
 	}
-	// Check if binning 2x2 is selected
 	else if (bin==2) 	{
-		// Select binning 2x2 (2028 x 1520 pixels)
-		roi = "--mode 2  --width 2028 --height 1520 ";
+		command += " --mode 2  --width 2028 --height 1520";
 	}
-	// Check if binning 3x3 is selected
 	else 	{
-		// Select binning 4x4 (1012 x 760 pixels)
-		roi = "--mode 4 --width 1012 --height 760 ";
+		command += " --mode 4 --width 1012 --height 760";
 	}
 
-	// Append binning window
-	command += roi;
-
 	if (asiExposure < 1)
 	{
 		asiExposure = 1;
 	}
-
-    // https://www.raspberrypi.org/documentation/raspbian/applications/camera.md : HQ (IMX477) 	200000000 (i.e. 200s)
-	if (asiExposure > 200000000)
+	else if (asiExposure > 200000000)
 	{
+		// https://www.raspberrypi.org/documentation/raspbian/applications/camera.md : HQ (IMX477) 	200000000 (i.e. 200s)
 		asiExposure = 200000000;
 	}
 
-	// Exposure time
-	string shutter;
-
 	// Check if automatic determined exposure time is selected
 
 	if (asiAutoExposure)
 	{
 		if (myModeMeanSetting.mode_mean) {
 			ss.str("");
 			ss << myRaspistillSetting.shutter_us;
-			shutter = "--exposure off --shutter " + ss.str() + " ";
+			command += " --exposure off";
+			command += " --shutter " + ss.str();
 		} else {
-			shutter = "--exposure auto ";
+			command += " --exposure auto";
 		}
 	}
 	// Set exposure time
 	else if (asiExposure)
 	{
 		ss.str("");
 		ss << asiExposure;
-		shutter = "--exposure off --shutter " + ss.str() + " ";
+		command += " --exposure off";
+		command += " --shutter " + ss.str();
 	}
 
-	// Add exposure time setting to raspistill command string
-	command += shutter;
-
 	if (asiAutoFocus)
 	{
-		command += "--focus ";
+		command += " --focus";
 	}
 
-	// Anolog Gain
-	string gain;
-
 	// Check if auto gain is selected
 	if (asiAutoGain)
 	{
 		if (myModeMeanSetting.mode_mean) {
 			ss.str("");
 			ss << myRaspistillSetting.analoggain;
-			gain = "--analoggain " + ss.str() + " ";
+			command += " --analoggain " + ss.str();
 
 			if (myRaspistillSetting.digitalgain > 1.0) {
```

Comment:
I haven't had any good experiences with digital gain - maybe we shouldn't support this parameter anymore.

Reply:
I would be fine with getting rid of it.  It may be a moot point anyway since libcamera doesn't support a separate digital gain argument - instead, anything over the maximum gain value is applied to digital gain.  However, I don't know what the maximum gain value is.